Formula/adios2.rb

```diff
@@ -0,0 +1,126 @@
+class Adios2 < Formula
+  desc "Next generation of ADIOS developed in the Exascale Computing Program"
+  homepage "https://adios2.readthedocs.io"
+  url "https://github.com/ornladios/ADIOS2/archive/v2.5.0.tar.gz"
+  sha256 "7c8ff3bf5441dd662806df9650c56a669359cb0185ea232ecb3578de7b065329"
+  head "https://github.com/ornladios/ADIOS2.git", :branch => "master"
```

Comment:
Is head needed? Or is upstream going to test regularly against homebrew? Or are the formula submitters testing and using it? If not, then it should be removed. Basically we want to include head if it's used regularly, especially to allow upstream to test, e.g., nightly.

Reply:
`master` is used pretty often by developers working directly with adios and rarely by applications as a dependency, so it depends on the audience.  I'm in the process now of adding macOS builds to our pr-driven CI so I'll be sure to have the specific homebrew configurations tested as part of that.

Reply:
Awesome! For OpenCoarrays I have nightly CI tests against master built via `brew install --HEAD opencoarrays`. (They often fail though, due to issues with Travis-CI... so choose your CI provider carefully.) I am strongly in favor of keeping `head` for this case. If you setup nightly testing with `brew install --HEAD adios` and `brew test --HEAD adios` (and maybe throw a `brew audit` in there for good measure) then it will:

- Notify you if changes to upstream dependencies break ADIOS
- Let you know if some subtle change broke ADIOS working correctly with Homebrew
- Make it easy/seamless for Homebrew version bumps of ADIOS.